DevEdu.API/DevEdu.Business.Tests/TaskServiceTests.cs

```diff
@@ -0,0 +1,349 @@
+using System;
+using System.Collections.Generic;
+using DevEdu.Business.Services;
+using DevEdu.DAL.Models;
+using DevEdu.DAL.Repositories;
+using Moq;
+using NUnit.Framework;
+
+namespace DevEdu.Business.Tests
+{
+    public class TaskServiceTests
+    {
+        private Mock<ITaskRepository> _taskRepoMock;
+        private Mock<ICourseRepository> _courseRepoMock;
+        private Mock<IStudentAnswerOnTaskRepository> _studentAnswerRepoMock;
+
+        [SetUp]
+        public void Setup()
+        {
+            _taskRepoMock = new Mock<ITaskRepository>();
+            _courseRepoMock = new Mock<ICourseRepository>();
+            _studentAnswerRepoMock = new Mock<IStudentAnswerOnTaskRepository>();
+        }
+
+
+        [Test]
+        public void AddTask_SimpleDtoWithoutTags_TaskCreated()
+        {
+            //Given
+            var expectedTaskId = 55;
+            var taskDto = new TaskDto { Name = "Task1", Description = "Description1", Links = "noLinks", IsRequired = true};
+
+            _taskRepoMock.Setup(x => x.AddTask(taskDto)).Returns(expectedTaskId);
+            _taskRepoMock.Setup(x => x.AddTagToTask(It.IsAny<int>(), It.IsAny<int>()));
+
+            var sut = new TaskService(_taskRepoMock.Object, _courseRepoMock.Object, _studentAnswerRepoMock.Object);
+
+            //When
+            var actualTaskId = sut.AddTask(taskDto);
+
+            //Than
+            Assert.AreEqual(expectedTaskId, actualTaskId);
+            _taskRepoMock.Verify(x => x.AddTask(taskDto), Times.Once);
+            _taskRepoMock.Verify(x => x.AddTagToTask(It.IsAny<int>(), It.IsAny<int>()), Times.Never);
+        }
+
+        [Test]
+        public void AddTask_DtoWithTags_TaskWithTagsCreated()
+        {
+            //Given
+            var expectedTaskId = 55;
+            var taskDto = new TaskDto {
```

Comment:
please create a static class (TaskData) and move this mock to a static method

please do it for all the methods of this class